stripe/src/main/java/com/stripe/android/StripePaymentController.kt

```diff
@@ -351,6 +351,12 @@ internal class StripePaymentController internal constructor(
         callback: ApiResultCallback<AlipayAuthResult>
     ) : ApiOperation<AlipayAuthResult>(callback = callback) {
         override suspend fun getResult(): AlipayAuthResult {
+            if (intent.paymentMethod?.liveMode == false) {
```

Comment:
Should it throw an exception or return a failure result?

Reply:
This is relying on the fact that ApiOperation will catch the IllegalArgumentException and return `Result.failure`. Definitely open to handling it a different way but this seemed like the most straight-forward given our current setup. That said, I'm not sure I love the current set up.

Reply:
Ship it